src/main/java/org/apache/maven/plugins/dependency/fromDependencies/BuildClasspathMojo.java

```diff
@@ -322,18 +326,20 @@ private boolean isUpdToDate( String cpString )
     }
 
     /**
-     * It stores the specified string into that file.
+     * Stores the specified string into that file.
      *
-     * @param cpString the string to be written into the file.
-     * @throws MojoExecutionException
+     * @param cpString the string to write into the file
      */
     private void storeClasspathFile( String cpString, File out )
         throws MojoExecutionException
     {
         // make sure the parent path exists.
         out.getParentFile().mkdirs();
+        
+        String encoding = getProject().getProperties().getProperty( "project.reporting.outputEncoding", "UTF-8" );
 
-        try ( Writer w = new BufferedWriter( new FileWriter( out ) ) )
+        try ( Writer w =
```

Comment:
Why the pain? `Files.newBufferedWriter(...)`?

Reply:
Old habits mostly, but looking at that it requires as much code to change types (File --> Path, String --> charset, boolean --> FileOption) that it's equally complex.